osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/endpoints/SmartMeteringInstallationEndpoint.java

```diff
@@ -369,4 +373,78 @@ public CoupleMbusDeviceByChannelResponse getCoupleMbusDeviceByChannelResponse(
+     * @return a response containing a correlationUid and the
+     *         deviceIdentification
+     * @throws OsgpException
+     */
+    @PayloadRoot(localPart = "DeCoupleMbusDeviceByChannelRequest", namespace = SMARTMETER_INSTALLATION_NAMESPACE)
+    @ResponsePayload
+    public DeCoupleMbusDeviceByChannelAsyncResponse deCoupleMbusDeviceByChannel(
+            @OrganisationIdentification final String organisationIdentification,
+            @RequestPayload final DeCoupleMbusDeviceByChannelRequest request,
+            @MessagePriority final String messagePriority, @ScheduleTime final String scheduleTime,
+            @ResponseUrl final String responseUrl) throws OsgpException {
+
+        final String deviceIdentification = request.getDeviceIdentification();
+        final short channel = request.getDeCoupleMbusDeviceByChannelRequestData().getChannel();
+        log.info("Incoming DeCoupleMbusDeviceByChannelRequest for device: {} and channel {}.", deviceIdentification,
+                channel);
+
+        DeCoupleMbusDeviceByChannelAsyncResponse response = null;
+        try {
+            response = new DeCoupleMbusDeviceByChannelAsyncResponse();
+
+            final String correlationUid = this.installationService.enqueueDeCoupleMbusDeviceByChannelRequest(
+                    organisationIdentification, deviceIdentification,
+                    MessagePriorityEnum.getMessagePriority(messagePriority),
+                    this.installationMapper.map(scheduleTime, Long.class), channel);
+
+            response.setCorrelationUid(correlationUid);
+            response.setDeviceIdentification(deviceIdentification);
+            this.saveResponseUrlIfNeeded(correlationUid, responseUrl);
+        } catch (final Exception e) {
+            log.error("Exception while cleaning up on channel: {} for device: {} for organisation {}.", channel,
+                    deviceIdentification, organisationIdentification, e);
+            this.handleException(e);
+        }
+        return response;
+    }
+
+    /**
+     * @param request
+     *            the request message containing the correlationUid
+     * @return the response message containing the OsgpResultType and optional a
+     *         message
+     * @throws OsgpException
+     */
+    @PayloadRoot(localPart = "DeCoupleMbusDeviceByChannelAsyncRequest", namespace = SMARTMETER_INSTALLATION_NAMESPACE)
+    @ResponsePayload
+    public DeCoupleMbusDeviceByChannelResponse getDeCoupleMbusDeviceByChannelResponse(
+            @RequestPayload final DeCoupleMbusDeviceByChannelAsyncRequest request) throws OsgpException {
+
+        DeCoupleMbusDeviceByChannelResponse response = null;
+        try {
+            final ResponseData responseData = this.responseDataService.dequeue(request.getCorrelationUid(),
+                    ComponentType.WS_SMART_METERING);
+
+            this.throwExceptionIfResultNotOk(responseData, "De Couple Mbus Device By Channel");
```

Comment:
seeing "De Couple.." here in an Exception message hurts my eyes and short circuits my brain.

Reply:
Decouple it is